pages/live/lobby.js

```diff
@@ -99,6 +135,40 @@ async function updateLobbyMap () {
 
 }
 
+/**
+ * Updates the UI to indicate whether we're the lobby host
+ */
+function updateLobbyHost () {
+
+  // Enable or disable settings buttons based on if we're the host
+  const buttons = [
+    document.querySelector("#lobby-name-button"),
+    document.querySelector("#lobby-password-button")
+  ];
+
+  if (amHost) {
+    for (const button of buttons) {
+      button.style.opacity = 1.0;
+      button.style.cursor = "pointer";
+      button.removeAttribute("onmouseover");
+      button.removeAttribute("onmouseleave");
+    }
+  } else {
+    for (const button of buttons) {
+      button.style.opacity = 0.5;
+      button.style.cursor = "default";
+      button.setAttribute("onmouseover", "showTooltip('Only the host can do this.')");
+      button.setAttribute("onmouseleave", "hideTooltip()");
+    }
+  }
+
+  // Hide the map select button entirely for non-hosts
+  const mapButton = document.querySelector("#lobby-map-button");
+  if (amHost) mapButton.style.display = "";
+  else mapButton.style.display = "none";
```

Comment:
According to my testing, this works fine up until a map is selected, at which point this part of the DOM is modified and the button becomes visible again